### PR TITLE
Output Solution

### PR DESCRIPTION
**Pull request by 1simjustin**
Will include certain refactoring changes in this PR as well

### CODE REVIEW

File: src/main/java/longah/commands/delete/DeleteCommand.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        switch (this.subCommand) {
+        case "member":
+            throw new LongAhException(ExceptionMessage.COMMAND_NOT_IMPLEMENTED);
+        case "transaction":
+            DeleteTransactionCommand deleteTransactionCommand =
+                    new DeleteTransactionCommand(fullCommandString, this.taskExpression);
+            deleteTransactionCommand.execute(group);
+            break;
+        default:
+            throw new LongAhException(ExceptionMessage.INVALID_DELETE_COMMAND);
```
Comment (djleong01) [inline src/main/java/longah/commands/delete/DeleteCommand.java:35]: delete [NUMBER] crashes the program

File: src/main/java/longah/commands/list/ListCommand.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+            ListMemberCommand listMemberCommand =
+                    new ListMemberCommand(fullCommandString, this.taskExpression);
+            listMemberCommand.execute(group);
+            break;
+        case "transactions":
+            ListTransactionCommand listTransactionCommand =
+                    new ListTransactionCommand(fullCommandString, this.taskExpression);
+            listTransactionCommand.execute(group);
+            break;
+        case "debt":
```
Comment (djleong01) [inline src/main/java/longah/commands/list/ListCommand.java:49]: change to "debts" to make it consistent with error message

**Comment by jing-xiang** [comment]
E2E
Testing logs are as follows:

Round 1:
image

members.txt
image

transactions.txt
image

Round 2:
list debts --> FAILED
image

members.txt
image

transactions.txt
image

Round 3:
image

members.txt
image

transactions.txt 
image

Round 4:

image

members.txt
image

transactions.txt
image


Summary:

add member NAME **[PASSED]**
add transaction LENDER p/BORROWER a/AMOUNT ....  **[PASSED]**
list transactions  **[PASSED]**
list debts  **[FAILED]**
list members  **[PASSED]**
find transactions NAME  **[PASSED]**
find debts NAME  **[PASSED]**
clear  **[PASSED]**
settleup NAME  **[PASSED]**
delete transaction INDEX  **[PASSED]**
exit  **[PASSED]**

File: src/main/java/longah/commands/delete/DeleteCommand.java (hunk 1 above)
Comment (jing-xiang) [reply src/main/java/longah/commands/delete/DeleteCommand.java:40]: Can confirm

File: src/main/java/longah/exception/ExceptionMessage.java [hunk 3]
```diff
@@ -15,27 +15,35 @@ public enum ExceptionMessage {
...
     STORAGE_FILE_NOT_READ ("File not read."),
     STORAGE_FILE_NOT_WRITTEN ("File not written."),
     INVALID_STORAGE_CONTENT ("Invalid content in storage file, line ignored."),
+    STORAGE_FILE_CORRUPTED ("Storage file is corrupted."),
 
     // Ui exceptions
-    INVALID_FINDPAYMENT_COMMAND("Invalid command format." +
-            " Use 'findPayment PERSON'"),
-    INVALID_SETTLEUP_COMMAND("Invalid command format." +
+    INVALID_COMMAND ("Invalid command. Use 'add', 'listdebts', 'listtransactions'," +
```
Comment (FeathersRe) [inline src/main/java/longah/exception/ExceptionMessage.java:31]: Good if the exception message for invalid command are updated to the new standards with command words separated

**Review by FeathersRe — COMMENTED** [review]
Minor error inconsistencies for exception messages

**Comment by djleong01** [comment]
image

can add a more specific error message where person specified is a lender rather than borrower

**Review by djleong01 — APPROVED** [review]
LGTM after testing and fixing all issues

**Comment by FeathersRe** [comment]
Testing of all commands works on my end except for finding debts NAME & find transactions NAME. 
1) Find debts includes entries where the member to search is a lender.
2) Find transactions include entries where the member to search is a borrower.

For instance,
Pre search commands:
image
Erroneous find transaction commands (Underlined in Red):
image
Erroneous find debts commands (Underlined in Red):
image

Would be good if such problems were solved to maintain consistency with previous program behaviors.

**Review by jing-xiang — APPROVED** [review]
LGTM

**Pull request merged**